stl/inc/chrono

```diff
@@ -652,6 +687,1103 @@ namespace chrono {
+        return day{static_cast<unsigned>(_Left) - _Right.count()};
+    }
+    constexpr days operator-(const day& _Left, const day& _Right) noexcept {
+        return days{static_cast<int>(static_cast<unsigned>(_Left)) - static_cast<int>(static_cast<unsigned>(_Right))};
+    }
+
+    class month {
+        unsigned char _Month;
+
+    public:
+        month() = default;
+        constexpr explicit month(unsigned _Val) noexcept : _Month{static_cast<unsigned char>(_Val)} {}
+
+        constexpr month& operator++() noexcept {
+            *this += months{1};
+            return *this;
+        }
+        constexpr month operator++(int) noexcept {
+            month _Temp{*this};
+            ++(*this);
+            return _Temp;
+        }
+        constexpr month& operator--() noexcept {
+            *this -= months{1};
+            return *this;
+        }
+        constexpr month operator--(int) noexcept {
+            month _Temp{*this};
+            --(*this);
+            return _Temp;
+        }
+
+        constexpr month& operator+=(const months& _Months) noexcept;
+        constexpr month& operator-=(const months& _Months) noexcept;
+
+        _NODISCARD constexpr explicit operator unsigned() const noexcept {
+            return static_cast<unsigned>(_Month);
+        }
+        _NODISCARD constexpr bool ok() const noexcept {
+            return _Month >= 1 && _Month <= 12;
+        }
+    };
+
+    constexpr bool operator==(const month& _Left, const month& _Right) noexcept {
+        return static_cast<unsigned>(_Left) == static_cast<unsigned>(_Right);
+    }
+    constexpr strong_ordering operator<=>(const month& _Left, const month& _Right) noexcept {
+        return static_cast<unsigned>(_Left) <=> static_cast<unsigned>(_Right);
+    }
+
+    constexpr month operator+(const month& _Left, const months& _Right) noexcept {
+        const auto _M   = static_cast<long long>(static_cast<unsigned>(_Left)) + (_Right.count() - 1);
+        const auto _Div = (_M >= 0 ? _M : _M - 11) / 12;
+        return month{static_cast<unsigned>(_M - _Div * 12 + 1)};
+    }
+    constexpr month operator+(const months& _Left, const month& _Right) noexcept {
+        return _Right + _Left;
+    }
+    constexpr month operator-(const month& _Left, const months& _Right) noexcept {
+        return _Left + -_Right;
+    }
+    constexpr months operator-(const month& _Left, const month& _Right) noexcept {
+        const auto _M = static_cast<unsigned>(_Left) - static_cast<unsigned>(_Right);
+        return months{_M <= 11 ? _M : _M + 12};
+    }
+
+    constexpr month& month::operator+=(const months& _Months) noexcept {
+        *this = *this + _Months;
+        return *this;
+    }
+    constexpr month& month::operator-=(const months& _Months) noexcept {
+        *this = *this - _Months;
+        return *this;
+    }
+
+    class year {
+        short _Year;
+        constexpr static int _Ymin = -32767;
+        constexpr static int _Ymax = 32767;

```

Comment:
Regarding naming - I think we accumulated some inconsistency when we converted macros to `constexpr` constants. New code typically uses `_Plain_ugly` identifiers for `constexpr` constants, since `_ALL_CAPS` looks like our macros. (Many examples in `<execution>`, `<charconv>`, `<ratio>`, etc.)

`_Year_min` and `_Year_max` seem like they would be nice and readable.

Reply:
I was looking at `_BUF_SIZE` in `xstring` 🤣